### ISSUE TITLE
Loading EPICS CA DLL failed on MAC - ARM (M2)

### ISSUE THREAD

**Issue by cappel89** [pyepics/pyepics#265]
It seems that we encounter a similar issue as in #264. After the release of 3.5.3, we were not able to use [Ophyd](https://github.com/bluesky/ophyd) on MAC, ARM (M2). We narrowed down the problem, and were able to reproduce it without Ophyd.

```python
from epics import ca
import ctypes
lib = ca.find_libca()
ctypes.cdll.LoadLibrary(lib)
```
The process gets killed, tested with Python 3.11.1. 
Unfortunately the fix in 3.5.4 does not solve our problem here.
Thanks in advance.

**Comment by newville** [comment]
@cappel89 I don't think that is similar to #264.  That is specifically about Windows and this is specifically about MacOS.  

I think this must be a problem with the dynamic library for Darwin arm64

**Comment by newville** [comment]
@cappel89 I can verify that the distributed darwinarm64 dylibs also fail - killing python - for me.  I know the files I had worked for me (and, as the README indicates, after doing a bit of `install_name_tool` surgery).   But now they don't and the files have different shasums.   I am not sure why this is - perhaps some mangling of the files in the GitHub commit process??

I re-copied from my base 7.0.7 build, and redid the `install_name_tool` changes, and I can load `libca.dylib` without crashing.  I just pushed this to the master branch,  including the shasums in the `clibs/darwinarm64/README` file.   

If you are willing to test that from the master branch, it would be helpful.

**Comment by wakonig** [comment]
I can confirm what @cappel89 experienced with 3.5.3 and 3.5.4 and it seems to be resolved on the current master branch. Thanks @newville for the quick fix!

**Comment by newville** [comment]
@wakonig @cappel89 Thanks, and yes, I also confirmed that running the tests with Darwin on arm64 (Python 3.11) worked for me on a Macbook with an M1 processor.   So that's re-assuring.

I am still a little unsure of what happened here.  Anyway, I'm happy to see this running natively on Darwin ARM64. 
But I will tag and release 3.2.5 soon -- later today.

**Comment by newville** [comment]
@cappel89 @wakonig OK, this should be fixed with the 3.5.5 release.  Thanks for your help and patience.

**Comment by cappel89** [comment]
@newville Thanks a lot for the quick response and smooth fix!
